### ISSUE TITLE
Add leafcutter ant leaf gathering

### ISSUE THREAD

**Issue by TimJentzsch** [Leafwing-Studios/Emergence#197]
Implement the first version of leafcutter ants.
See #188 for more details.

The ants should have an inventory with a maximal capacity.
When their inventory is empty, they should search for an acacia plant (see #196) and gather leaves. Gathering a leaf can be instant for now (if it's easier) and should transfer the leaf from the plant's output to the ant's inventory.
Once the ant inventory is filled, the ant should walk around again.

Actually using the leaves will be done in a separate issue, to keep the scope small.

Blocked by #196.

**Comment by alice-i-cecile** [comment]
This is fixed!
